tests/gdb-tests/tests/test_command_canary.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+
+import gdb
+import pytest
+
+import pwndbg.aglib.memory
+import pwndbg.aglib.regs
+import tests
+
+CANARY_X86_64_BINARY = tests.binaries.get("canary.x86-64.out")
+CANARY_I386_BINARY = tests.binaries.get("canary.i386.out")
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "binary, reg_name",
+    [
+        (CANARY_X86_64_BINARY, "rax"),
+        (CANARY_I386_BINARY, "eax"),
+    ],
+    ids=["x86-64", "i386"],
+)
+def test_command_canary(start_binary, binary, reg_name):
+    """
+    Tests the canary command for x86-64 and i386 architectures
+    """
+    start_binary(binary)
+
+    gdb.execute("break main")
+    gdb.execute("run")
+    gdb.execute("stepi")
+
+    register = getattr(pwndbg.aglib.regs, reg_name)
+    canary_value, at_random = pwndbg.commands.canary.canary_value()
```

Comment:
Fwiw shouldn't this test actually do sth like `output = gdb.execute("canary", to_string=True)` and assert on that?